src/main/java/edu/kis/powp/jobs2d/command/manager/MacroCommandManager.java

```diff
@@ -0,0 +1,121 @@
+package edu.kis.powp.jobs2d.command.manager;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.observer.Publisher;
+
+/**
+ * Driver command Manager.
+ */
+public class MacroCommandManager implements Job2dDriver {
+    private DriverCommand currentCommand = null;
+    private boolean macroState=false;
+    private List<String> names=new ArrayList<>();
+    private Publisher changePublisher = new Publisher();
+    List<DriverCommand> commands = new ArrayList<DriverCommand>();
+    List<List<DriverCommand>> macro= new ArrayList<>();
+    /**
+     * Set current command.
+     *
+     * @param commandList Set the command as current.
+     */
+    public synchronized void setCurrentCommand(DriverCommand commandList) {
+        this.currentCommand = commandList;
+        changePublisher.notifyObservers();
+    }
```

Comment:
This class has too many responsibilities. There is already DriverCommandManager available which we can use to set any command and execute it ("Run command"). Macro feature duplicates. 
Please reuse DriverCommandManager in this responsibility and simplify MacroCommandManager (the name should probably be changed after that).